smarts/core/traffic_history_provider.py

```diff
@@ -65,65 +75,71 @@ def sync(self, provider_state):
         # Ignore other sim state
         pass
 
-    def step(self, provider_actions, dt, elapsed_sim_time) -> ProviderState:
-        timestamp = min(
-            (
-                float(ts)
-                for ts in self._traffic_history_service.all_timesteps
-                if float(ts) >= elapsed_sim_time
-            ),
-            default=None,
-        )
-        if not self._traffic_history_service or timestamp is None:
-            return ProviderState(vehicles=[])
+    class _HistoryRow(NamedTuple):
+        vehicle_id: int
+        vehicle_type: int
+        vehicle_length: float
+        vehicle_width: float
+        position_x: float
+        position_y: float
+        heading_rad: float
+        speed: float
 
-        time_with_offset = str(round(timestamp + self.start_time_offset, 1))
-        if not self._traffic_history_service.fetch_history_at_timestep(
-            time_with_offset
-        ):
-            return ProviderState(vehicles=[])
+    @staticmethod
+    def _decode_vehicle_type(vehicle_type):
+        # Options from NGSIM and INTERACTION currently include:
+        #  1=motorcycle, 2=auto, 3=truck, 4=pedestrian/bicycle
+        # But we don't yet have glb models for 1 and 4.
+        if vehicle_type == 3:
+            return "truck"
+        return "passenger"
```

Comment:
I see this as a potential issue that we would default to a passenger vehicle when provided a pedestrian since we do not support non-road actors yet nor do we yet want to evaluate pedestrians.

Reply:
Yeah, I agree.  I'm adding a change to skip pedestrians for now.

Reply:
... although on second thought, I think we should probably just add a pedestrian model fairly soon because, in an imitation learning scenario, it could cause problems to train from a vehicle that swerves or stops suddenly for seeming no reason (if a pedestrian that triggered this behavior is not part of the simulation).  
So I guess I'm not going to skip them after all.  We should just do Issue #756 relatively soon instead.

Reply:
Yes, more-so I was worried about imitation learning attempting to take over a pedestrian because it thinks it is a passenger vehicle.

Reply:
Ah oh.  I just added something to prevent that:  only passenger cars will now be included in the agent missions used by `history_vehicle_replacement_for_imitation_learning.py`.